examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -53,22 +53,34 @@ def get_api_url(self) -> str:
         return f"{self.auth_backend.get_webserver_url()}/api/v1"
 
     def list_dags(self) -> List["DagInfo"]:
-        response = self.auth_backend.get_session().get(f"{self.get_api_url()}/dags")
-        if response.status_code == 200:
-            dags = response.json()
-            webserver_url = self.auth_backend.get_webserver_url()
-            return [
-                DagInfo(
-                    webserver_url=webserver_url,
-                    dag_id=dag["dag_id"],
-                    metadata=dag,
-                )
-                for dag in dags["dags"]
-            ]
-        else:
-            raise DagsterError(
-                f"Failed to fetch DAGs. Status code: {response.status_code}, Message: {response.text}"
+        results = []
+        offset = 0
+        while True:
+            response = self.auth_backend.get_session().get(
+                f"{self.get_api_url()}/dags",
+                params={"limit": 100, "offset": offset, "order_by": "dag_id"},
             )
+            if response.status_code == 200:
+                dags = response.json()
+                webserver_url = self.auth_backend.get_webserver_url()
+
+                cur_results = [
+                    DagInfo(
+                        webserver_url=webserver_url,
+                        dag_id=dag["dag_id"],
+                        metadata=dag,
+                    )
+                    for dag in dags["dags"]
+                ]
+                results.extend(cur_results)
+                if len(cur_results) < 100:
+                    break
+            else:
+                raise DagsterError(
+                    f"Failed to fetch DAGs. Status code: {response.status_code}, Message: {response.text}"
+                )
+            offset += 100
+        return results
```

Comment:
seems like it should be a separate change

Reply:
Yea this was part of debugging, i was running into an infuriating version error with airflow that I since resolved, but this seemed like a good change so I left it in. I can place it in a followup.